src/test/groovy/com/bmuschko/gradle/docker/tasks/image/DockerfileTest.groovy

```diff
@@ -2,52 +2,97 @@ package com.bmuschko.gradle.docker.tasks.image
 
 import spock.lang.Specification
 
+import java.util.logging.Level
+import java.util.logging.Logger
+
 import static com.bmuschko.gradle.docker.tasks.image.Dockerfile.*
 
 class DockerfileTest extends Specification {
+    private static Logger LOG = Logger.getLogger(DockerfileTest.class.getCanonicalName());
     def "Instruction String representation is built correctly"() {
+
         expect:
-        instructionInstance.keyword == keyword
-        instructionInstance.build() == builtInstruction
+        LOG.fine "testing " + [
+            instruction: instructionInstance.class,
+            becomes: [
+                keyword: expectedKeyword,
+                withBuildText: expectedBuiltInstruction,
+            ]
+        ]
+
+        def actual
+        try {
+            actual = instructionInstance.build()
+        }
+        catch(Exception e) {
+            LOG.info "exception catched with message `${e.message}`"
+            if(!expectedBuiltInstruction.equals(e.class)) {
+                LOG.log(Level.WARNING, e.stackTrace?.join('\n\t'))
+            }
+            actual = e.class
+        }
+
+        instructionInstance.keyword == expectedKeyword
+        actual == expectedBuiltInstruction
```

Comment:
1. I had to catch the exception thrown in build. The expected type of exception is defined in `expectedBuiltInstruction`. If it doesn't match, the assertion fails like:
```
actual == expectedBuiltInstruction
|      |  |
|      |  ""
|      false
class java.lang.IllegalArgumentException
```
Without the catch, the exception would be thrown and will not produce a clean spock test result assertion

2. the LOG will show the stacktrace of the exception, if it's not the expected type to get some informations about what is going wrong .. indeed ... we might remove that

Last but not least: I'll change the message from Germish to English ;)